ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/internal/PeerDiscoveryController.java

```diff
@@ -586,6 +590,17 @@ public void setRetryDelayFunction(final RetryDelayFunction retryDelayFunction) {
     this.retryDelayFunction = retryDelayFunction;
   }
 
+  public void handleBondingRequest(final DiscoveryPeer peer) {
+    final DiscoveryPeer peerToBond =
+        peerTable.get(peer).filter(known -> known.discoveryEndpointMatches(peer)).orElse(peer);
```

Comment:
Shouldn't we also check against the `bondingPeers` collection?  

Maybe we need a helper method `resolvePeer` that takes a peer and replaces it with representations from the `peerTable` or `bondingPeers` collection as needed.  Then we could dedupe the same logic in `onMessage`.